src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -486,6 +520,86 @@ public function update( $type, $item ) {
 			}
 		}
 
+		if ( 'theme' === $type ) {
+			error_log( '    Theme ' . var_export( $item->theme, true ) . ' has been upgraded.' );
+		}
+
+		if ( 'plugin' === $type ) {
+			error_log( '    Plugin ' . var_export( $item->slug, true ) . ' has been upgraded.' );
+			if ( is_plugin_inactive( $upgrader_item ) ) {
+				error_log( '    ' . var_export( $upgrader_item, true ) . ' is inactive and will not be checked for fatal errors.' );
+			}
+
+			if ( $was_active && ! is_wp_error( $upgrade_result ) ) {
+
+				/*
+				 * The usual time limit is five minutes. However, as a loopback request
+				 * is about to be performed, increase the time limit to account for this.
+				 */
+				if ( function_exists( 'set_time_limit' ) ) {
+					set_time_limit( 10 * MINUTE_IN_SECONDS );
+				}
+
+				// Avoid a race condition when there are 2 sequential plugins that have fatal errors.
```

Comment:
```suggestion
				/*
				 * Avoids a race condition when there are 2 sequential plugins that have fatal errors.
				 * WP_Automatic_Updater::update() can be called for the second plugin before the first has
				 * completed WP_Automatic_Updater::has_fatal_error() skipping has_fatal_error() for the
				 * second, and possibly subsequent plugins.
				 */
```

Reply:
@johnbillion I think this has a better explanation.

Reply:
Slight modification:
```suggestion
				/*
				 * Avoids a race condition when there are 2 sequential plugins that have fatal errors.
				 * WP_Automatic_Updater::update() can be called for the second plugin before the first has
				 * finished checking for fatal errors. This can cause the second plugin's fatal error checking to be skipped, and may affect subsequent plugins too.
				 */
```

Reply:
@johnbillion what do you think of this one?

Reply:
I think I still need to tweak this a bit. After testing locally with sequentially fataling plugins I found that without a 2 sec delay before the loopback check, the check does not fail, I assume it has something to do with loading the new plugin into memory rather than anything else. Very strange, but at least the length of time for the loopback shouldn't be an issue.

I tried using `wp_opcache_invalidate_directory()` instead of the sleep but that failed too.

I also tried a `sleep(1)` but that failed.

Looks like `sleep(2)` it is.